src/core/SparseMatrix.cpp

```diff
@@ -68,6 +68,15 @@ SparseMatrix::SetZeros(){
     fArmaMat = arma::sp_mat(fNCols, fNRows);
 }
 
+void
+SparseMatrix::SetIdentity(){
+    if(!fNRows && !fNCols && fNCols!=fNRows )
+        return;
+    for (size_t i = 0; i < fNCols; ++i) {
+        fArmaMat(i,i)=1;
```

Comment:
that's true, but because you're not creating a new matrix you don't know if the off diagonals have been previously accounted for and remain non-zero